src/main/java/de/tum/in/www1/artemis/service/metis/conversation/GroupChatService.java

```diff
@@ -0,0 +1,120 @@
+package de.tum.in.www1.artemis.service.metis.conversation;
+
+import static javax.validation.Validation.buildDefaultValidatorFactory;
+
+import java.util.Set;
+
+import javax.validation.ConstraintViolationException;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.metis.ConversationParticipant;
+import de.tum.in.www1.artemis.domain.metis.conversation.GroupChat;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationParticipantRepository;
+import de.tum.in.www1.artemis.repository.metis.conversation.GroupChatRepository;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.metis.conversation.dtos.GroupChatDTO;
+
+@Service
+public class GroupChatService {
+
+    public static final String GROUP_CHAT_ENTITY_NAME = "messages.groupchat";
+
+    private static final String GROUP_NAME_REGEX = "^[a-z0-9-]{0,20}$";
+
+    private final UserRepository userRepository;
+
+    private final ConversationParticipantRepository conversationParticipantRepository;
+
+    private final ConversationService conversationService;
+
+    private final GroupChatRepository groupChatRepository;
+
+    public GroupChatService(UserRepository userRepository, ConversationParticipantRepository conversationParticipantRepository, ConversationService conversationService,
+            GroupChatRepository groupChatRepository) {
+        this.userRepository = userRepository;
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.conversationService = conversationService;
+        this.groupChatRepository = groupChatRepository;
+    }
+
+    /**
+     * Gets a group chat by its id from the database or throws an exception if it does not exist
+     *
+     * @param groupChatId the id of the group chat
+     * @return the group chat
+     */
+    public GroupChat getGroupChatOrThrow(Long groupChatId) {
+        return groupChatRepository.findById(groupChatId)
```

Comment:
Same as above.

Reply:
I do not like to break the Resource -> Service -> Repository layer that often tbh. I think its not "bad" to have it in the service. So I do not need to add a new dependency to the resource

Reply:
Done in a213247889b860e291776ad1627056bcd7eb77c0